src/txn/txn_ckpt.c

```diff
@@ -1948,6 +1948,16 @@ __wt_checkpoint_close(WT_SESSION_IMPL *session, bool final)
         (!F_ISSET(S2C(session), WT_CONN_FILE_CLOSE_SYNC) && !metadata)))
         return (__wt_set_return(session, EBUSY));
 
+    /*
+     * Checkpoint shouldn't occur when final is set to false, without either checkpoint lock or
+     * schema lock. Potential race condition arises when backup is ongoing, causing inconsistencies
+     * between the versions of data.
```

Comment:
The assert itself is fine. I think the comment needs more information. Maybe replace with something like:
```
/*
 * Make sure there isn't a potential race between backup copying the metadata and a checkpoint changing the metadata.
 * Backup holds both the checkpoint and schema locks. Checkpoint should hold those also except on the final checkpoint
 * during close. Confirm the caller either is the final checkpoint or holds at least one of the locks.
 * /
 ```